zipline/__main__.py

```diff
@@ -179,7 +176,6 @@ def _(*args, **kwargs):
 @click.option(
     '--trading-calendar',
     metavar='TRADING-CALENDAR',
-    type=click.Choice(default_calendar_names),
```

Comment:
I kind of like including all of the options here. Looking at `TradingCalendarDispatcher`, it seems like we're actually indirectly defining what we want here in the `has_calendar` method:
https://github.com/quantopian/zipline/blob/49a3a4a2785a4a75dc21f04389094681eb408579/zipline/utils/calendars/calendar_utils.py#L92-L100

What would you think of splitting out an `available_calendar_names` property that just returns the union of what we check against in `has_calendar`? Then we could use that in the definition of `has_calendar`, and also expose it for the CLI.

Reply:
Have we loaded all the user extensions by this point in time, e.g. at zipline import time?

Reply:
Oh good point, not necessarily 😞 

Probably worth just leaving this out then.